xray/test/test_dataset.py

```diff
@@ -937,6 +937,30 @@ def test_align(self):
         with self.assertRaises(TypeError):
             align(left, right, foo='bar')
 
+    def test_broadcast(self):
+        ds = Dataset({'foo': 0, 'bar': ('x', [1]), 'baz': ('y', [2, 3])},
+                     {'c': ('x', [4])})
+        expected = Dataset({'foo': (('x', 'y'), [[0, 0]]),
+                            'bar': (('x', 'y'), [[1, 1]]),
+                            'baz': (('x', 'y'), [[2, 3]])},
+                            {'c': ('x', [4])})
+        actual, = broadcast(ds)
+        self.assertDatasetIdentical(expected, actual)
+
+        ds_x = Dataset({'foo': ('x', [1])})
+        ds_y = Dataset({'bar': ('y', [2, 3])})
+        expected_x = Dataset({'foo': (('x', 'y'), [[1, 1]])})
+        expected_y = Dataset({'bar': (('x', 'y'), [[2, 3]])})
+        actual_x, actual_y = broadcast(ds_x, ds_y)
```

Comment:
Never mind the previous comment. It looks like you test coordinate broadcasting here.